spec/requests/borrowing_request_spec.rb

```diff
@@ -0,0 +1,67 @@
+require "rails_helper"
+
+RSpec.describe BorrowingsController, type: :controller do
+  let(:user) {FactoryBot.create :user, id: 1}
+  let(:category) {FactoryBot.create :category}
+  let(:author) {FactoryBot.create :author}
+  let(:publisher) {FactoryBot.create :publisher}
+
+  let!(:book) do
+    FactoryBot.create :book, category_id: category.id,
+                             publisher_id: publisher.id,
+                             author_id: author.id,
+                             quantity_borrowed: 8
+  end
+
+  let!(:borrowing_two) do
+    FactoryBot.create :borrowing, user_id: user.id,
+                                  borrow_code: "XATPVCZEBM",
+                                  status: :pending
+  end
+  let!(:borrowing_three) do
+    FactoryBot.create :borrowing, user_id: user.id,
+                                  borrow_code: "XATPVCZEBM",
+                                  status: :accept,
+                                  date_borrow: "2021-01-28 17:23:02",
+                                  date_pay: "2021-02-03 17:23:02"
+  end
+
+  let(:borrowing_params) {FactoryBot.attributes_for :borrowing}
+
+  before {login user}
+
+  describe "GET #index" do
+    before {get :index, params: {page: 1}}
+
+    it "should render the 'index' template with @borrowings" do
+      aggregate_failures do
+        expect(response).to render_template :index
+        expect(assigns(:borrowings)).to eq [borrowing_two, borrowing_three]
+      end
+    end
+  end
+
+  describe "POST #create" do
+    before do
+      post :create, params: {user_id: user.id,
+      borrowing:{
+        borrow_items_attributes: [
+          {book_id: book.id}
+        ]
+      }}
+    end
+
+    it "should create new borrowing" do
+      aggregate_failures do
```

Comment:
aggregate_failures là ntn e nhỉ

Reply:
Lý thuyết là một API cho phép nhóm một tập hợp expect và xem tất cả các thất bại cùng một lúc, thay vì nó bỏ dở trong lần thất bại đầu tiên. 
Thì em hiểu aggregate_failures cho phép nhiều expect trong khối được cung cấp không thành công và sau đó tổng hợp chúng thành một ngoại lệ duy nhất, thay vì hủy bỏ expect đầu tiên không thành công như bình thường. Điều này cho phép xem tất cả các thất bại từ toàn bộ tập hợp các expect mà không cần tách từng lỗi ra thành các it riêng ( điều bày có thể làm chậm mọi thứ nếu it phức tạp).